src/test/java/com/amazon/opendistroforelasticsearch/sql/esintgtest/SQLFunctionsIT.java

```diff
@@ -159,6 +160,74 @@ public void caseChangeWithAggregationTest() throws IOException {
                 hitAny("/aggregations/UPPER_1/buckets", kvString("/key", equalTo("AMBER"))));
     }
 
+    @Test
```

Comment:
Could you add the typical tests that broke the code in past? 

1. Aggregation

SELECT CAST(age as INT) intage FROM index GROUP BY intage

2. ORDER BY 
SELECT CAST(age as INT) intage FROM index ORDER BY intage

3. something crazier, like nested casts

SELECT cast(CAST (1 AS varchar) as float)

(that one could be ignored, because I don't expect it to work, but at least we will see how the code behaves).

Reply:
Good suggestion- definitely will add